scripts/helpers/deploy.js

```diff
@@ -193,91 +197,108 @@ async function deployWithoutProxy(nameInState, artifactName, deployer, construct
   const netId = await web3.eth.net.getId()
   const state = readNetworkState(network.name, netId)
 
-  console.log(`Deploying ${artifactName}...`)
-  if (constructorArgs.length > 0) {
-    console.log(`Constructor args: ${constructorArgs}`)
-  }
+  process.stdout.write(`Deploying ${artifactName} (NO proxy)... `)
 
   const contract = await deployContract(artifactName, constructorArgs, deployer)
-  console.log(`${artifactName} (NO proxy): ${contract.address}`)
 
   const gasUsed = await getDeploymentGasUsed(contract)
-  console.log(`    gas used ${gasUsed}`)
+  console.log(`done: ${contract.address} (gas used ${gasUsed})`)
   TOTAL_GAS_USED += gasUsed
 
-  if (nameInState) {
-    persistNetworkState2(network.name, netId, state, {
-      [nameInState]: {
-        "contract": artifactName,
-        [addressFieldName]: contract.address,
-        "constructorArgs": constructorArgs,
-      }
-    })
+  state[nameInState] = {
+    ...state[nameInState],
+    contract: await getContractPath(artifactName),
+    [addressFieldName]: contract.address,
+    constructorArgs: constructorArgs,
   }
+  persistNetworkState(network.name, netId, state)
 
+  console.log()
   return contract.address
 }
 
+async function deployImplementation(nameInState, artifactName, deployer, constructorArgs=[]) {
+  const netId = await web3.eth.net.getId()
+  const state = readNetworkState(network.name, netId)
+
+  process.stdout.write(`Deploying implementation for proxy of ${artifactName}... `)
+  const contract = await deployContract(artifactName, constructorArgs, deployer)
+
+  const gasUsed = await getDeploymentGasUsed(contract)
+  TOTAL_GAS_USED += gasUsed
+
+  console.log(`done: ${contract.address} (gas used ${gasUsed})`)
+
+  state[nameInState] = { ...state[nameInState] }
+  state[nameInState].implementation = {
+    contract: await getContractPath(artifactName),
+    address: contract.address,
+    constructorArgs: constructorArgs,
+  }
+  persistNetworkState2(network.name, netId, state)
```

Comment:
persistNetworkState2 -> persistNetworkState ?

Reply:
removed duplicate function, left the more straightforward version of it